### PR TITLE
feat: switch to weighted caboose

### PR DESCRIPTION
**Pull request by lidel**
Taking https://github.com/filecoin-saturn/caboose/pull/26 for a spin, will rebase and  merge after upstream is finalized.

cc @aarshkshah1992  @hacdias (for visibility)

### CODE REVIEW

File: blockstore.go [hunk 1]
```diff
@@ -87,16 +87,20 @@ func newBlockStore(orchestrator, loggingEndpoint string) (blockstore.Blockstore,
...
+	}
+
 	return caboose.NewCaboose(&caboose.Config{
 		OrchestratorEndpoint: *oe,
-		OrchestratorClient:   http.DefaultClient,
+		OrchestratorClient:   saturnServiceClient,
 
 		LoggingEndpoint: *le,
-		LoggingClient:   http.DefaultClient,
+		LoggingClient:   saturnServiceClient,
```
Comment (lidel) [inline blockstore.go:99]: :information_source: @guanzo I think we've found the offender – we were using custom user-agent client only for saturn client, orchestrator, and logs used  default one. This PR (when merged) should fix this.

File: blockstore.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ func (e *exchangeBsWrapper) GetBlock(ctx context.Context, c cid.Cid) (blocks.Blo
 	defer cancel()
 
 	if goLog.Level().Enabled(zapcore.DebugLevel) {
-		goLog.Debugw("block requested from saturn", "cid", c.String())
+		goLog.Debugw("block requested from strn", "cid", c.String())
```
Comment (lidel) [inline blockstore.go:37]: :information_source: consistent with caboose errors

Comment (willscott) [reply blockstore.go:37]: (should we spell it out?)

**Review by willscott — APPROVED** [review]
This all looks good.

We should probably get the caboose PR merged to main, but happy to do that separately

File: blockstore.go (hunk 2 above)
Comment (lidel) [reply blockstore.go:37]: I mean, someone else made this decision: https://strn.pl/ipfs/bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi :upside_down_face:

**Pull request merged**